tests/typecheck/managers/querysets/test_basic_methods.yml

```diff
@@ -70,12 +70,75 @@
 
 -   case: queryset_method_of_union
     main: |
-        from typing_extensions import reveal_type
+        from typing_extensions import TypedDict, reveal_type
+        from django.db import models
         from myapp.models import MyModel1, MyModel2
         kls: type[MyModel1 | MyModel2] = MyModel1
         reveal_type(kls.objects)  # N: Revealed type is "django.db.models.manager.Manager[myapp.models.MyModel1] | django.db.models.manager.Manager[myapp.models.MyModel2]"
         reveal_type(kls.objects.all())  # N: Revealed type is "django.db.models.query.QuerySet[myapp.models.MyModel1, myapp.models.MyModel1] | django.db.models.query.QuerySet[myapp.models.MyModel2, myapp.models.MyModel2]"
         reveal_type(kls.objects.get())  # N: Revealed type is "myapp.models.MyModel1 | myapp.models.MyModel2"
+
+        # Regular QuerySet
+        foos = MyModel1.objects.all()
+        bars = MyModel2.objects.all()
```

Comment:
Can we also please test structurally incompatible models? For now we don't do anything special with them.

Reply:
I can add a todo but yes we won't detect much currently (but runtime will most likely see errors like `The used SELECT statements have a different number of columns`)

Reply:
Good enough for now! 🤝

Reply:
Done !